packages/legend-query-builder/src/stores/explorer/QueryBuilderExplorerState.ts

```diff
@@ -114,7 +117,9 @@ export abstract class QueryBuilderExplorerTreeNodeData implements TreeNodeData {
     mappingData: QueryBuilderExplorerTreeNodeMappingData,
   ) {
     makeObservable(this, {
```

Comment:
@MauricioUyaguari or @YannanGao-gs , do you know why we don't include `isOpen`/`setIsOpoen` as properties in this `makeObservable` call? I tested out adding those and it made it so I wouldn't have to call `this.refreshTree()` on line 745 in the `highlightTreeNode` function (https://github.com/finos/legend-studio/pull/2969/files#diff-f03db7e81110555697c5eeec710ace3f7123cc5f1eed7004008bff10c70b7843R745). However, I didn't want to make this change just in case we intentionally left out `isOpen`/`setIsOpen` here.

Reply:
we may have overlooked this. however it may also be the case, that when we go through the processes of finding a node, we want to open the nodes all at once instead of opening them while recursively looking for them. we can look to enable this but we should look at places where we open an unopened node deep into the tree and make sure that works as expected.